application/src/main/java/org/opentripplanner/updater/trip/siri/SiriRealTimeTripUpdateAdapter.java

```diff
@@ -156,30 +156,43 @@ private Result<UpdateSuccess, UpdateError> apply(
     } catch (DataValidationException e) {
       return DataValidationExceptionMapper.toResult(e, journey.getDataSource());
     } catch (Exception e) {
-      LOG.warn(
-        "{} EstimatedJourney {} failed.",
-        shouldAddNewTrip ? "Adding" : "Updating",
-        DebugString.of(journey),
-        e
-      );
+      LOG.warn("{} EstimatedJourney {} failed.", siriUpdateType, DebugString.of(journey), e);
       return Result.failure(UpdateError.noTripId(UNKNOWN));
     }
   }
 
-  /**
-   * Check if VehicleJourney is a replacement departure according to SIRI-ET requirements.
-   */
-  private boolean shouldAddNewTrip(
+  private SiriUpdateType updateType(
     EstimatedVehicleJourney vehicleJourney,
     EntityResolver entityResolver
   ) {
-    // Replacement departure only if ExtraJourney is true
-    if (!(TRUE.equals(vehicleJourney.isExtraJourney()))) {
-      return false;
+    // Extra call if at least one of the call is an extra call
+    if (
+      ((vehicleJourney.getRecordedCalls() != null &&
+          vehicleJourney
+            .getRecordedCalls()
+            .getRecordedCalls()
+            .stream()
+            .anyMatch(recordedCall -> Boolean.TRUE.equals(recordedCall.isExtraCall()))) ||
+        (vehicleJourney.getEstimatedCalls() != null &&
+          vehicleJourney
+            .getEstimatedCalls()
+            .getEstimatedCalls()
+            .stream()
+            .anyMatch(estimatedCall -> Boolean.TRUE.equals(estimatedCall.isExtraCall()))))
+    ) {
```

Comment:
Probaby beyond the scope of the PR but what do you think of extracting these very wordy calls to vehicleJourney into a re-usable wrapper type?

Reply:
There are a few other pieces of code that could be moved into it?

Reply:
You're right, I added the isExtraCall() method in CallWrapper and forgot to refactor this piece of code afterwards.
Done.